src/jit/decomposelongs.cpp

```diff
@@ -920,64 +924,319 @@ GenTree* DecomposeLongs::DecomposeShift(LIR::Use& use)
 
     GenTree* tree   = use.Def();
     GenTree* gtLong = tree->gtGetOp1();
+    GenTree* oldShiftByOp = tree->gtGetOp2();
+
     genTreeOps oper = tree->OperGet();
+    genTreeOps shiftByOper = oldShiftByOp->OperGet();
 
     assert((oper == GT_LSH) || (oper == GT_RSH) || (oper == GT_RSZ));
 
     LIR::Use loOp1Use(Range(), &gtLong->gtOp.gtOp1, gtLong);
-    loOp1Use.ReplaceWithLclVar(m_compiler, m_blockWeight);
+    unsigned loOp1LclNum = loOp1Use.ReplaceWithLclVar(m_compiler, m_blockWeight);
```

Comment:
Don't we need to check whether lop1 is already an LclVar before replacing with another LclVar?

Same comment applies to line 938